plugins/live_stream_stats/lib/LiveStreamStatsActions.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * @package api
+ * @subpackage cache
+ */
+
+require_once(dirname(__FILE__) . '/../../../api_v3/lib/KalturaResponseCacher.php');
+
+/**
+ * Class LiveStreamStatsActions
+ *
+ * Package and location is not indicated
+ * Should not include any kaltura dependency in this class - to enable it to run in cache only mode
+ */
+class LiveStreamStatsActions
+{
+	const ENTRY_ID_ARG = 'entryId';
+	const CONF_LIVE_STATS_INTERVAL = 'liveStatsInterval';
+	const LIVE_SETTINGS = 'live';
+	const DEFAULT_TTL = 20;
+
+	/**
+	 * @var LiveStreamStatsCacheHandler
+	 */
+	protected $liveStreamStatsCacheHandler;
+
+	/**
+	 * @var int
+	 */
+	protected $liveStreamStatsCacheTTL;
+
+	/**
+	 *
+	 * @throws Exception
+	 */
+	public function __construct()
+	{
+		$this->liveStreamStatsCacheHandler = null;
+		$this->liveStreamStatsCacheTTL = 0;
+
+		$this->liveStreamStatsCacheTTL = kConf::get(LiveStreamStatsActions::CONF_LIVE_STATS_INTERVAL,LiveStreamStatsActions::LIVE_SETTINGS, LiveStreamStatsActions::DEFAULT_TTL);
+
+		$cache = kCacheManager::getSingleLayerCache(kCacheManager::CACHE_TYPE_PLAYS_VIEWS);
+		if (!$cache)
+		{
+			throw new Exception("Could not initiate cache instance (needed for Live Stream Stats)");
+		}
+
+		$this->liveStreamStatsCacheHandler = new LiveStreamStatsCacheHandler($this->liveStreamStatsCacheTTL, $cache);
+	}
+
+	protected function responseHandlingGetStats(): void
+	{
+		KalturaResponseCacher::setExpiry($this->liveStreamStatsCacheTTL);
+		KalturaResponseCacher::setHeadersCacheExpiry($this->liveStreamStatsCacheTTL);
+	}
+
+	public function doGetLiveStreamStats(string $entryId): int
+	{
+		$this->responseHandlingGetStats();
+		return $this->liveStreamStatsCacheHandler->getNumberOfViewers($entryId);
+	}
+
+	public static function getLiveStreamStats_validate($params)
+	{
+		if (is_null($params))
+		{
+			return 'Missing parameter for getLiveStreamStats action';
+		}
+

```

Comment:
why not actually validating the params? (entryId)

Reply:
because we dont really care. it's better to return 0 if the entryId is not in the table, than to make a call to verify.